### DISCUSSION TITLE
How does acme-companion read other containers' environment variables?

### DISCUSSION THREAD

**Discussion by AlperShal** [nginx-proxy/acme-companion#995] [Q&A] [answered]
I just can't stop thinking about this. Also I couldn't find anything on Docker's docs. How can acme read other containers' VIRTUAL_HOST environment variables and create certificates for them? Or did I understand the system completely wrong?

**Comment by buchdag** [comment, marked as answer]
Hi @AlperShal.

acme-companion reads the other containers environments by running [docker-gen](https://github.com/nginx-proxy/docker-gen) internally (like nginx-proxy).

docker-gen it self does this through the [Docker API](https://docs.docker.com/engine/api/), and access that API through the Docker socket.

**Reply by AlperShal** [reply]
Thanks for the answer!
